### PR TITLE
WellKnownNames

### PR DESCRIPTION
**Pull request by jansule**
**Important:** Please wait with merging this PR until

- [x] [this PR from geostyler-style](https://github.com/terrestris/geostyler-style/pull/43)
was merged.

Update parser to parse WellKnownNames. geostyler-sld-parser now is able to parse WellKnownNames according to definitions in above mentioned PR. 

Added tests and testfiles for every supported WellKnownName. Updated existing tests and testfiles to match new definitions.

### CODE REVIEW

File: data/styles/point_simplecross.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
```
Comment (dnlkoch) [inline data/styles/point_simplecross.ts]: Minor: Please remove empty line.

File: data/xml2jsObjects/point_simplecross.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+}
```
Comment (dnlkoch) [inline data/xml2jsObjects/point_simplecross.json]: Missing newline.

File: data/xml2jsObjects/point_simplesquare.json [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+}
```
Comment (dnlkoch) [inline data/xml2jsObjects/point_simplesquare.json]: Missing newline.

File: data/xml2jsObjects/point_simplestar.json [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+}
```
Comment (dnlkoch) [inline data/xml2jsObjects/point_simplestar.json]: Missing newline.

File: data/xml2jsObjects/point_simpletriangle.json [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+}
```
Comment (dnlkoch) [inline data/xml2jsObjects/point_simpletriangle.json]: Missing newline.

File: data/xml2jsObjects/point_simplex.json [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                  }
+                ]
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+}
```
Comment (dnlkoch) [inline data/xml2jsObjects/point_simplex.json]: Missing newline.

File: src/SldStyleParser.ts [hunk 7]
```diff
@@ -975,55 +1072,72 @@ class SldStyleParser implements StyleParser {
...
           '$': {
             'name': 'stroke'
           }
         });
       }
-      if (circleSymbolizer.strokeWidth) {
+      if (markSymbolizer.strokeWidth) {
         strokeCssParameters.push({
-          '_': circleSymbolizer.strokeWidth,
+          '_': JSON.stringify(markSymbolizer.strokeWidth),
```
Comment (dnlkoch) [inline src/SldStyleParser.ts]: Why not simply `toString()` instead?

File: src/SldStyleParser.ts [hunk 8]
```diff
@@ -975,55 +1072,72 @@ class SldStyleParser implements StyleParser {
...
           '$': {
             'name': 'stroke-width'
           }
         });
       }
+      if (markSymbolizer.strokeOpacity) {
+        strokeCssParameters.push({
+          '_': JSON.stringify(markSymbolizer.strokeOpacity),
```
Comment (dnlkoch) [inline src/SldStyleParser.ts]: See above.

File: src/SldStyleParser.ts [hunk 9]
```diff
@@ -975,55 +1072,72 @@ class SldStyleParser implements StyleParser {
...
+    if (markSymbolizer.radius) {
+      graphic[0].Size = [JSON.stringify(markSymbolizer.radius)];
+    }
+
+    if (markSymbolizer.opacity) {
+      graphic[0].Opacity = [JSON.stringify(markSymbolizer.opacity)];
+    }
+
+    if (markSymbolizer.rotate) {
+      graphic[0].Rotation = [JSON.stringify(markSymbolizer.rotate)];
```
Comment (dnlkoch) [inline src/SldStyleParser.ts]: See above.

**Comment by jansule** [comment]
@terrestris/devs ready to merge.

File: src/SldStyleParser.ts [hunk 10]
```diff
@@ -975,55 +1072,72 @@ class SldStyleParser implements StyleParser {
...
     }
 
     let graphic: any[] = [{
       'Mark': mark
     }];
-    if (circleSymbolizer.radius) {
-      graphic[0].Size = circleSymbolizer.radius;
+
+    if (markSymbolizer.radius) {
+      graphic[0].Size = [JSON.stringify(markSymbolizer.radius)];
```
Comment (ahennr) [inline src/SldStyleParser.ts]: Can `toString()` be used here as well?

File: src/SldStyleParser.ts [hunk 11]
```diff
@@ -975,55 +1072,72 @@ class SldStyleParser implements StyleParser {
...
+    }
+
+    if (markSymbolizer.opacity) {
+      graphic[0].Opacity = [markSymbolizer.opacity.toString()];
+    }
+
+    if (markSymbolizer.rotate) {
+      graphic[0].Rotation = [JSON.stringify(markSymbolizer.rotate)];
```
Comment (ahennr) [inline src/SldStyleParser.ts]: Can `toString()` be used here as well?

**Pull request merged**